### PR TITLE
Add fade to all vfx

### PR DESCRIPTION
**Pull request by EdwardRegasDuncan**
Updated Emissive VFX to use fade-out on:
Player take damage
Player heal
enemy takes damage
boss takes damage

### CODE REVIEW

**Review by Bernatmago — APPROVED** [review]
Everything working good, I would make the boss hit effect transition shorter

File: Gameplay/entities/enemies/BossController.h [hunk 1]
```diff
@@ -299,6 +299,9 @@ namespace Hachiko
             std::vector<EnemyController*> enemies;
             float enemy_timer = 0.0;
 
+            float damage_effect_duration = 1.0f;
```
Comment (baransrc) [inline Gameplay/entities/enemies/BossController.h:302]: Let's make this a serialized field for ease of editing.

File: Gameplay/entities/enemies/EnemyController.cpp [hunk 2]
```diff
@@ -785,7 +791,7 @@ void Hachiko::Scripting::EnemyController::RegisterHit(int damage, float3 directi
 	}
 
 	_combat_stats->ReceiveDamage(damage);
-	game_object->ChangeEmissiveColor(float4(255, 255, 255, 255), 0.3f, true);
+	damage_effect_progress = damage_effect_duration;
```
Comment (baransrc) [inline Gameplay/entities/enemies/EnemyController.cpp:794]: If damage_effect_duration is higher than 1.0 this won't work, we should be setting the effect process to 1.0 if we are using the decrement method (damage_effect_progress -= Time::DeltaTime() / damage_effect_duration) (i would make it reverse so it goes from 0 to 1 though).

File: Gameplay/entities/enemies/EnemyController.h [hunk 3]
```diff
@@ -188,6 +188,9 @@ namespace Hachiko
             SERIALIZE_FIELD(float, _current_spawning_time);
             bool _has_spawned = false;
 
+            float damage_effect_duration = 1.0f;
```
Comment (baransrc) [inline Gameplay/entities/enemies/EnemyController.h:191]: Same applies to here, this would be better off serialized.

**Review by baransrc — CHANGES_REQUESTED** [review]
Thanks for the good work!! We just need couple of changes, and then we are good to go!!

**Pull request merged**